### PR TITLE
drivers/l3g4200d : Expose Configurations to Kconfig

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in L3G4200D Sensor Device driver to Kconfig.

### Testing procedure
1. New documentation was built using doxygen 

The build works fine.

2. New macro was introduced in tests/driver_l3g4200d/main.c for testing.

```
#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))
```
Firmware was uploaded to FIT IoT Lab test bed.

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-311-g04c32-Kconfig_l3g4200d_tests)
CONFIG_L3G4200D_DEFAULT_ADDRESS=0x68
L3G4200 gyroscope driver test application

Initializing L3G4200 sensor
[Failed]

#### Usage with CFLAGS 

/tests/driver_l3g4200d/Makefile

> CFLAGS += -DCONFIG_L3G4200D_DEFAULT_ADDRESS=0x69

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-311-g04c32-Kconfig_l3g4200d_tests)
CONFIG_L3G4200D_DEFAULT_ADDRESS=0x69
L3G4200 gyroscope driver test application

Initializing L3G4200 sensor
[Failed]

#### Usage with Kconfig

/tests/driver_l3g4200d/

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-311-g04c32-Kconfig_l3g4200d_tests)
CONFIG_L3G4200D_DEFAULT_ADDRESS=0x69
L3G4200 gyroscope driver test application

Initializing L3G4200 sensor
[Failed]

Note : The sensor is not available for interfacing hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri  Please merge this only after  #13981

### CODE REVIEW

File: drivers/include/l3g4200d.h [hunk 1]
```diff
@@ -44,13 +44,13 @@
 /**
  * @brief Default address
  *
- * The address depends on the status of SDO Pin.
- * Default address corresponds to SD0 connected to GND.
- * For more information on SerialBus Address, refer Section 5.1.1
- * I2C Operation on datasheet.
+ * The address depends on the status of SDO Pin. Default
+ * address corresponds to SDO connected to GND. For more
+ * information refer section 'I2C Operation' in the
```
Comment (leandrolanzieri) [inline drivers/include/l3g4200d.h:49]: Was the section number wrong here?

Comment (akshaim) [reply drivers/include/l3g4200d.h:49]: Section number was right. I noticed that there could be multiple versions of datasheet, section title helps than the section number.

**Comment by leandrolanzieri** [comment]
Please rebase

**Comment by akshaim** [comment]
> Please rebase

Done

**Comment by leandrolanzieri** [comment]
Please rebase

**Comment by akshaim** [comment]
> Please rebase

Done.

**Pull request merged**